lib/src/over_react_redux/hooks/use_selector.dart

```diff
@@ -0,0 +1,183 @@
+// Copyright 2021 Workiva Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+@JS()
+library over_react_redux.hooks.use_selector;
+
+import 'package:js/js.dart';
+import 'package:over_react/src/over_react_redux/over_react_redux.dart';
+import 'package:over_react/src/over_react_redux/hooks/use_dispatch.dart';
+import 'package:over_react/src/util/context.dart';
+import 'package:react/react_client/react_interop.dart' show ReactContext;
+
+// ----------------------------------------------------
+//  useSelector hook
+// ----------------------------------------------------
+
+/// A hook that allows you to extract data from the Redux `Store` state, using a [selector] function.
+///
+/// The use of this hook will also subscribe your component to the Redux `Store`, and run your [selector] whenever
+/// an action is dispatched.
+///
+/// * The [selector] should be pure since it is potentially executed multiple times and at arbitrary points in time.
+/// * The [selector] is approximately equivalent to the `mapStateToProps` argument of [connect] conceptually.
+/// * The [selector] will be called with the entire Redux `Store` state as its only argument.
+/// * The [selector] will be run whenever the function component renders.
+///
+/// By default, the return value of `selector` is compared using strict (`===`) equality. If you want to
+/// customize how equality is defined, pass a comparator function to the [equalityFn] argument.
+///
+/// If you need to use a selector with custom [Context], use [createSelectorHook] instead.
+///
+/// See the [react-redux JS documentation](https://react-redux.js.org/api/hooks#useselector) for more details.
+///
+/// ### Example
+///
+/// > This example assumes that your `Counter` component is rendered as the descendant of a [ReduxProvider] component
+///   that is wired up to a Redux `Store` instance with a `CounterState` instance containing a field called `count`.
+///
+/// ```dart
+/// import 'package:over_react/over_react.dart';
+/// import 'package:over_react/over_react_redux.dart';
+/// import 'counter_state.dart';
+///
+/// mixin CounterProps on UiProps {}
+///
+/// UiFactory<CounterProps> Counter = uiFunction(
+///   (props) {
+///     final count = useSelector<int, CounterState>((state) => state.count);
```

Comment:
I feel like having the generic parameters the other way around feels more natural:
```dart
// Current:
final count = useSelector<int, CounterState>((state) => state.count);
// My preference
final count = useSelector<CounterState, int>((state) => state.count);
```

I'd be curious to see what other people think, though.